pkg/openwsn/doc.txt

```diff
@@ -0,0 +1,218 @@
+ * will simply drop frames with invalid crc. The stack can function correctly if
+ * radio drivers automaticaly drop frames with an invalid crc (i.e. the stack
+ * doesn't get notified about these frames), but it might print the following
+ * error if using `openwsn_serial`:
+ *
+ * - `[IEEE802154E] wdDataDuration overflows while at state 19 in slotOffset 0`
+ *
+ * ### uart
+ *
+ * In RIOT, the first configured UART device is mapped to STDIO in most cases.
+ * In OpenWSN however, the `openserial` tool uses UART to feed external software
+ * running on a host computer such as
+ * [Openvisualizer](https://github.com/openwsn-berkeley/openvisualizer).
+ * To enable use of these tools, a UART adaptation is provided.
+ *
+ * By default when `openwsn_serial` (`openserial`) is used STDIO will be disabled
+ * (it will use `stdio_null`). When multiple uart are available STDIO and `openserial`
+ * can be used in parallel.
+ *
+ * OpenWSN uart abstraction makes use of tx hardware interrupts to execute a
+ * previously registered callback after every byte is sent out. These interrupts
+ * are currently not defined in RIOT in a generic way, so instead a timer is to
+ * fire shortly after a byte is written until sw_isr are possible.
+ *
+ * This implementation uses `ztimer` since its already used as an abstraction
+ * for `periph_rtt`.
+ *
+ * ## Tested Platforms and Pin configurations
+ *
+ * So far, this has been successfully tested on `iotlab-m3`,`nucleo-f103` and
+ * `samr21-xpro`, all based on at86rf23x radios.
+ *
+ * ### Timing
+ *
+ * Timing is essential for OpenWSN to work properly. For optimal results most
+ * parameters in `board_info.h` should be measured for the specific hardware used.
+ * OpenWSN has done that for most of their boards. These values can not be taken
+ * directly from OpenWSN since they do not necessarily use the same TIMER's or
+ * clock speeds.
+ *
+ * For more details on those parameters refer to:
+ * https://openwsn.atlassian.net/wiki/spaces/OW/pages/688251/State+Machine
+ *
+ * Since all these parameters are HW dependant, it also means that hybrid
+ * networks (different type of underlying hardware) might desynchronize often,
+ * or not manage to keep in sync at all.
+ *
+ * ### ledpins & debugpins
+ *
+ * The OpenWSN software provides different hooks all over the stack to toggle
+ * different LEDs as well as debug pins to examine state and scheduling of a node.
+ * Default configuration files are provided for both. The LED configuration maps to
+ * RIOTs `LEDX_PIN` definitions, if available. On Nucleo boards LED0 line is
+ * shared SPI, so is not used.
+ * The default configuration can be overwritten by setting `OPENWSN_LEDPINS_DEFAULT`
+ * in the form of `ledpins_config_t`. The debugpins work similarly by setting
+ * `OPENWSN_DEBUGPINS_BOARD` in the form of `debugpins_config_t`.
+ *
+ * The default configuration maps to OpenWSN reference hardware `openmote-b`.
```

Comment:
Yes, I was hopping to support cc2538, by now but I still have issues. It still makes sense to me to match this configuration since the hardware has the exact amount of pins and leds.